### ISSUE TITLE
Possibility of wrong exit code message when smoke testing multiple implementations

### ISSUE THREAD

**Issue by sudo-jarvis** [bowtie-json-schema/bowtie#913]
Identified a bug in which when we smoke test multiple implementations, if some prior implementation gives an error then for all next implementations: `❌ some failures` would be printed since the `exit_code` variable isnt being reinitialized to 0 after each implementation.

```
    exit_code = 0

    for implementation in implementations:
        echo(f"Testing {implementation.name!r}...\n", file=sys.stderr)
        serializable: list[dict[str, Any]] = []

        async for _, results in implementation.smoke():
            async for case, result in results:
                if result.unsuccessful():
                    exit_code |= _EX_DATAERR

                match format:
                    case "json":
                        serializable.append(
                            dict(
                                case=case.without_expected_results(),
                                result=asdict(result.result),
                            ),
                        )

                    case "pretty":
                        echo(f"  · {case.description}: {result.dots()}")

                    case "markdown":
                        echo(f"* {case.description}: {result.dots()}")

        match format:
            case "json":
                echo(json.dumps(serializable, indent=2))

            case "pretty":
                message = "❌ some failures" if exit_code else "✅ all passed"
                echo(f"\n{message}", file=sys.stderr)

            case "markdown":
                message = (
                    "**❌ some failures**"
                    if exit_code
                    else "**✅ all passed**"
                )
                echo(f"\n{message}", file=sys.stderr)

    return exit_code
```

**Comment by Julian** [comment]
No that I agree is a bug -- I thought I'd filed it somewhere, I'd definitely noticed it previously -- that message *is* meant to be per implementation, so we shouldn't print it for every implementation after one fails, it should appear only under an implementation which has failures.

**Comment by Julian** [comment]
Fixed in #914.
